### PR TITLE
Sample operator

### PR DESCRIPTION
**Pull request by jakub-dzon**
Adds reference operator implementation.

Signed-off-by: Jakub Dzon <jdzon@redhat.com>

### CODE REVIEW

File: examples/sample-operator/README.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+```
+
+Deployment of the HTTP server is controlled by the presence of a `sampleconfig` CR; to deploy it, execute:
+```shell script
+kubectl apply -f deploy/crd/sampleconfig_types.go
+```
+
+The HTTP service listens on port 8081, which can be forwarded locally:
+```shell script
+kubectl -n kubevirt port-forward service/http-server 8081:8081
```
Comment (pkliczewski) [inline examples/sample-operator/README.md:20]: What do you think about adding this in the operator as any other resource we deploy?

File: examples/sample-operator/deploy/operator.yaml [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+            - name: POD_NAME
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.name
+            - name: OPERATOR_NAME
+              value: "sample-operator"
+            - name: OPERATOR_VERSION
+              value: v0.0.3
+            - name: SERVER_IMAGE
+              value: quay.io/jdzon/sample-http-server:v0.0.3
```
Comment (pkliczewski) [inline examples/sample-operator/deploy/operator.yaml:196]: We could add http nodePort service definition here.

File: examples/sample-operator/pkg/controller/sampleconfig/cr-manager.go [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+	serviceAccount.Namespace = namespace
+
+	role := resourceBuilder.CreateRole(httpServerName, []rbacv1.PolicyRule{})
+	role.Namespace = namespace
+
+	roleBinding := resourceBuilder.CreateRoleBinding(httpServerName, httpServerName, httpServerName, namespace)
+	roleBinding.Namespace = namespace
+
+	httpDeployment := m.createHTTPServerDeployment()
+	httpService := m.createHTTPServerService()
```
Comment (pkliczewski) [inline examples/sample-operator/pkg/controller/sampleconfig/cr-manager.go:68]: We could add http service definition here.

Comment (jakub-dzon) [reply examples/sample-operator/pkg/controller/sampleconfig/cr-manager.go:68]: :+1: I will expose the HTTP server port on the node in the server service definition.

**Pull request merged**